api_specs/v2/component_analyses.yaml

```diff
@@ -1,17 +1,101 @@
 openapi: 3.0.1
 info:
-  title: Analytics Core API Specs for Version 2
-  version: "2"
+  title: Analytics Core API Specs
+  version: "2.1"
 servers:
 - url: https://f8a-analytics-2445582058137.production.gw.apicast.io/api/v2
   description: "Dependency Analytics Server"
 security:
 - tokenAuth: []
 paths:
+  /component-analyses/:
+    post:
+      tags:
+        - CA batch call
+      summary: Get security and other details about a software component and recommendations to change to a more secured version in a batched manner.
+      parameters:
+        - name: uuid
+          in: header
+          description: User Identification uuid
+          required: false
+          schema:
+            type: string
+      requestBody:
+        description: CA Batch Request
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                ecosystem:
+                  type: string
+                  example: pypi
+                packageversions:
+                  type: array
+                  items:
+                    type: object
+                    properties:
+                      package:
+                        type: string
+                        example: django
+                      version:
+                        type: string
+                        example: 1.1
+              example:
+                ecosystem: "pypi"
+                packageversions:
+                  - package: "django"
+                    version : "1.1"
+                  - package: "requests"
+                    version : "2.1"
+
+      responses:
+        200:
+          description: Detailed response for the requested application component batch
+          headers:
+            uuid:
+              schema:
+                type: string
+              description: User uuid.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  anyOf:
+                    - $ref: '#/components/schemas/FreeUser'
+                    - $ref: '#/components/schemas/RegisteredUser'
+                    - $ref: '#/components/schemas/NoVulnerabilities'
+              examples:
+                FreeUser Batch:
+                  $ref: '#/components/examples/free-user-batch'
+                RegisteredUser Batch:
+                  $ref: '#/components/examples/registered-user-batch'
+                No Vulnerability Batch:
+                  $ref: '#/components/examples/no-vul-batch'
+
+        202:
```

Comment:
Is it only for all? What if only 1/10 is a unknown?

Reply:
202: If all are unknown.

Now if 1/10 is unknown, we simply trigger ingestion for this. No notification to the user is circled back.

Reply:
It is too much of specialisation to have 202 only if all requests are unknown. Better to stick with 200 then. CC: @prashbnair

Reply:
Don't think sticking with 200 is good idea at all.

Reply:
In this case, it should be 202 if we do not send all the packages back in the response. 
In the case of the GET api, if we are not able to return then it should be 404.

Reply:
Ok, got it. If no. of incoming packages are not equal to output.. we will have 202. Okay